### PR TITLE
Test $HOME when it's parent is bind mounted with --userns=keep-id

### PR DESCRIPTION
**Pull request by debarshiray**
When --userns=keep-id is used, Podman is supposed to set up the home
directory of the user inside the container to match that on the host
as long as the home directory or any of its parents are marked as
volumes to be bind mounted into the container.

Currently, the test only considers the case where the home directory
itself is bind mounted into the container. It doesn't cover the Podman
code that walks through all the bind mounts looking for ancestors in
case the home directory itself wasn't specified as a bind mount.

Therefore, this improves the existing test added in commit
6ca8067956128585 ("Setup HOME environment when using --userns=keep-id")

Signed-off-by: Debarshi Ray <rishi@fedoraproject.org>

### CODE REVIEW

File: test/e2e/toolbox_test.go [hunk 1]
```diff
@@ -371,10 +372,21 @@ var _ = Describe("Toolbox-specific testing", func() {
...
 		Expect(session.OutputToString()).To(ContainSubstring(currentUser.HomeDir))
+
+		location := path.Dir(currentUser.HomeDir)
+		volumeArg := fmt.Sprintf("%s:%s", location, location)
+		session = podmanTest.Podman([]string{"run",
+			"--userns=keep-id",
+			"--volume", volumeArg,
+			fedoraToolbox, "sh", "-c", "echo $HOME"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
```
Comment (TomSweeneyRedHat) [inline test/e2e/toolbox_test.go:388]: This line looks to be failing in the CI....

Comment (debarshiray) [reply test/e2e/toolbox_test.go]: Yes, the test doesn't seem to like being run as root.

**Review by giuseppe — APPROVED** [review]
LGTM

**Comment by vrothberg** [comment]
Thanks, @debarshiray !

**Comment by giuseppe** [comment]
> /lgtm cancel

was it supposed to be /hold cancel? :-)

**Comment by vrothberg** [comment]
> > /lgtm cancel
> 
> was it supposed to be /hold cancel? :-)

I really need to grab a coffee

**Comment by giuseppe** [comment]
> I really need to grab a coffee

double please :-D

/hold cancel

**Comment by debarshiray** [comment]
Thanks for the reviews!

**Pull request merged**